docs/intermediate/least_to_most.md

```diff
@@ -0,0 +1,114 @@
+
+<div style={{textAlign: 'center'}}>
+  <img src={leastToMost} style={{width: "300px"}}
+        alt="A diagram of a least to most prompt showing how the prompt is composed of multiple steps whose answers are concatenated with the next step" />
+</div>
+
+<div style={{textAlign: 'center'}}>
+   Diagram of a Least to Most prompt
+</div>
+
+This leads to multiple improvements:
+- improved accuracy over Chain of Thought
+- increased generalization on problems harder than those in the prompt
+
+## Example: letter concatenation
+
+Let's examine the problem of concatenating the last letter of individual words (for example, given "cake, etymology" as input words, the output should be "ey").
+
+### First attempt: Standard 
+
+The standard prompt with few-shot examples performs very poorly, even with a more advanced model such as text-davinci-003.
+
+<div trydyno-embed="" openai-model="text-davinci-003"
+     initial-prompt="Q: think, machine\nA: ke\n\nQ: learning, reasoning, generalization\nA: ggn\n\nQ: artificial, intelligence\nA: le\n\nQ: transformer, language, vision\nA: ren\n\nQ: foo,bar,baz,blip\nA:"
+     initial-response="lip"
+     max-tokens="256" box-rows="18"
+     model-temp="0.2" ></div>
+
+### Second attempt: Chain of Thought
+Chain of Thought is significantly better. This is because it now allows the model
+to consider extracting the last letter of each word on its own, reducing the complexity down to the operation of grouping letters it has previously collected. However, this starts to break down at larger sizes.
+
+<div trydyno-embed="" openai-model="text-davinci-003"
+     initial-prompt="Q: think, machine\nA: The last letter of &#34;think&#34; is &#34;k&#34;. The last letter of &#34;machine&#34; is &#34;e&#34;. So &#34;think, machine&#34; is &#34;ke&#34;.\n\nQ: learning, reasoning, generalization\nA: The last letter of &#34;learning&#34; is &#34;g&#34;. The last letter of &#34;reasoning&#34; is &#34;n&#34;. The last letter of &#34;generalization&#34; is &#34;n&#34;. So &#34;learning, reasoning, generalization&#34; is &#34;ggn&#34;.\n\nQ: artificial, intelligence\nA: The last letter of &#34;artificial&#34; is &#34;l&#34;. The last letter of &#34;intelligence&#34; is &#34;e&#34;. So &#34;artificial, intelligence&#34; is &#34;le&#34;.\n\nQ: transformer, language, vision\nA: The last letter of &#34;transformer&#34; is &#34;r&#34;. The last letter of &#34;language&#34; is &#34;e&#34;. The last letter of &#34;vision&#34; is &#34;n&#34;. So &#34;transformer, language, vision&#34; is &#34;ren&#34;.\n\nQ: foo,bar,baz,blip\nA:"
+     initial-response="The last letter of &#34;foo&#34; is &#34;o&#34;. The last letter of &#34;bar&#34; is &#34;r&#34;. The last letter of &#34;baz&#34; is &#34;z&#34;. The last letter of &#34;blip&#34; is &#34;p&#34;. So &#34;foo,bar,baz,blip&#34; is &#34;orzp&#34;."
+     max-tokens="256" box-rows="18"
+     model-temp="0.2" ></div>
+
+### Third attempt: Least to Most
```

Comment:
How is this least to most? Isn't it just CoT?

Reply:
It is, it's meant to illustrate what exactly the difference is between the two styles, which took me a bit to understand. I thought that progression was the best to illustrate.

Reply:
Are you saying it is CoT? The section title says `Least to Most`

Reply:
Oh I see, this section is least to most because it will go "think,machine" -> k+e -> ke, "think,machine,reasoning" -> ke + g -> keg, so it reuses the answer of the previous step. That's the subtle part that took me a little bit to understand, and here the reduction into multiple subproblems is implicit. These are the prompts they use in the paper appendix.

Reply:
I probably should put exactly that sentence into the article, in fact. My first instinct was also "huh, isn't this just CoT?" But it really shows in longer steps, because it basically just "reduces" (like a monoid, or fold() in programming terms) the input, instead of having to keep attention to the entire chain to come up with the answe.

Reply:
done.